pulumi/pulumi-aws
# Schematize lambda.CallbackFunction overlay and add docs for it

The CallbackFunction resource overlay was not present in the registry before. This change adds the schema for it and adds the docs for the registry.
The schema is mostly based on the `lambda.Function` resource with modifications to adhere to the typescript class here: https://github.com/pulumi/pulumi-aws/blob/7461d6834b90936a38e80cc4bac2c10ff97499a0/sdk/nodejs/lambda/lambdaMixins.ts#L245

### How to review?
- Check out this PR locally
- Check out https://github.com/pulumi/registry and run the following in the root of that repository:
  - Follow the setup steps here: https://github.com/pulumi/registry?tab=readme-ov-file#installing-dependencies
  - `go install github.com/pulumi/registry/tools/resourcedocsgen@master`
  - `resourcedocsgen docs --schemaFile ../pulumi-aws/provider/cmd/pulumi-resource-aws/schema.json --docsOutDir content/registry/packages/aws/api-docs --packageTreeJSONOutDir static/registry/packages/navs/aws.json --version 6.0.0-alpha.0+dev` (you might have to tweak the path to the schema.json to match your dir layout)
  - `make serve`
- Visit http://localhost:1313/registry/packages/aws/api-docs/lambda/callbackfunction

Fixes #4351
